ocaml/nbd/src/main.ml

```diff
@@ -177,14 +177,10 @@ let main port certfile _ =
   in
   (* Log unexpected exceptions *)
   let () =
-    Lwt_main.run
-      (Lwt.catch t (fun e ->
-           Lwt_log.fatal_f "Caught unexpected exception: %s"
-             (Printexc.to_string e)
-           >>= fun () -> Lwt.fail e
-       )
-      )
+    try Lwt_main.run (ignore_exn_delayed t ())
```

Comment:
This is correct I think: Lwt_main.run according to its documentation will raise an exception if the promise failed, and the try/with will also catch any OCaml exceptions that might escape.

You might also want to set the async exception hook https://ocsigen.org/lwt/latest/api/Lwt#VALasync_exception_hook, which by default prints the exception and terminates the program.